### DISCUSSION TITLE
useInfiniteQueries example with skip instead of cursor

### DISCUSSION THREAD

**Discussion by JulienDemarque** [trpc/trpc#1307] [Q&A] [answered]
Got a use-case where using prisma cursor isn't fitting, can i use the endpoint cursor input as a skip property instead...?

**Comment by KATT** [comment, marked as answer]
Yup, I've done that myself, just have something like `cursor: z.number().min(0).default(0)`

It can also act as a page if you do `limit*cursor`

**Comment by JulienDemarque** [comment]
<img width="489" alt="Screen Shot 2021-11-25 at 13 25 09" src="https://user-images.githubusercontent.com/27313954/143433144-d0f22c73-7f0c-477e-a482-1f848e3dabea.png">
Got an other issue with that, ...data.pages seems to be an array of either 
`BaseRun[] | { items: BaseRun[]; nextCursor: number; }`
Why is that?

**Comment by KATT** [comment]
I need to see your procedure to answer that

**Comment by JulienDemarque** [comment]
Here is my definition:
```
  .query('findByBaseId', {
    input: z.object({
      baseId: z.string().uuid(),
      skip: z.number(),
      take: z.number(),
      sort: z.record(z.string().nullish()).nullish(),
      cursor: z.number().nullish(), // we are not using cursor for prisma, only to tell if there are more baseRuns to be fetched
    }),
    resolve: async ({ input }) => {
      // check if we are sorting variables
      if (keys(input.sort).length > 0 && keys(input.sort)[0] !== 'deadline') {
        const allBaseRuns = await prisma.baseRun.findMany({
          where: {
            baseId: input.baseId,
          },
        })
        const baseVariableName = keys(input.sort)[0]
        let sortedBaseRuns = sortBy((br: BaseRun) =>
          flow(
            get('variables'),
            values,
            find((v) => v.baseVariable.name === baseVariableName),
            get('value')
          )(br)
        )(allBaseRuns)
        if (input.sort?.[baseVariableName] === 'desc') {
          sortedBaseRuns = sortedBaseRuns.reverse()
        }
        const subsetSortedBaseRuns = slice(
          input.skip,
          input.skip + input.take
        )(sortedBaseRuns)
        return subsetSortedBaseRuns
      }
      const items = await prisma.baseRun.findMany({
        where: {
          baseId: input.baseId,
        },
        skip: input.skip,
        take: input.take,
        orderBy: {
          deadline: (input.sort?.deadline || undefined) as
            | Prisma.SortOrder
            | undefined,
        },
      })
      return {
        items,
        nextCursor: input.skip + 50,
      }
    },
  })
```  
 and the consumer:
 ```
   const baseRunData = useInfiniteQuery([
    'baseRun.findByBaseId',
    {
      baseId: base.id,
      take: 50,
      skip: baseViewOptions.skipRange,
      sort: {
        ...sortDeadlineOptions,
        ...sortVariableOptions,
      },
    },
  ])
 ```

**Comment by JulienDemarque** [comment]
Also struggling to infer the type from the query:
<img width="1095" alt="Screen Shot 2021-11-25 at 13 46 46" src="https://user-images.githubusercontent.com/27313954/143436224-8b687167-2ca6-40c8-a999-7659e314ed67.png">

**Comment by JulienDemarque** [comment]
If you have access to our repo: https://github.com/invisible-tech/invisible/blob/5780f864b099e2563c52c872b264090cad62c598/apps/manticore/components/Base.tsx#L269

https://github.com/invisible-tech/invisible/blob/5780f864b099e2563c52c872b264090cad62c598/libs/ultron/trpc/server/src/lib/routers/baseRun.ts#L32

**Comment by KATT** [comment]
Looks like you do an early return where you don't return `nextCursor`

**Comment by JulienDemarque** [comment]
Oh yes thank you! sorry about that, my bad...

**Comment by JulienDemarque** [comment]
@KATT sorry to poke you again, have you ever tried to use getQueryData to get the infinite query cache?  can't seem to get it to work...
